src/lib/colorUtils.ts

```diff
@@ -9,3 +17,41 @@ export const mapColorsToScheme = <T>(
     { ...siteVars.emphasisColors, ...siteVars.naturalColors },
     typeof mapper === 'number' ? String(mapper) : (mapper as any),
   ) as ColorValues<T>
+
+export const getColorSchemeFn = <T>(colorProp: string, colorScheme: ColorValues<T>) => {
+  const colors = _.get(colorScheme, colorProp)
+  return (area: keyof T, defaultColor: string) => (colors ? colors[area] : defaultColor)
+}
+
+export const getColorSchemeFromObject = (
+  colorScheme: ColorValues<Partial<ColorScheme>>,
+  colors: ComplexColorPropType,
+): Partial<ColorScheme> =>
+  _.mapValues(colors, (color, colorName) => {
+    // if the color scheme contains the color, then get the value from it, otherwise return the color provided
+    const colorSchemeValue = _.get(colorScheme, color, colorScheme.default[color])
+    return colorSchemeValue ? colorSchemeValue[colorName] : colors[colorName]
+  })
+
+export const getColorSchemeWithCustomDefaults = (
+  colorScheme: ColorSchemeMapping,
+  customDefaultValues: Partial<ColorScheme>,
+) => {
+  const mergedDefaultValues = {
```

Comment:
in future see this being reduced to a single concept of 'merging color schemes' - this should simplify the scheme object itself, make merging logic simpler and remove `default` prop